packages/strapi-plugin-users-permissions/services/Providers.js

```diff
@@ -161,6 +162,23 @@ const getProfile = async (provider, query, callback) => {
         });
       break;
     }
+    case 'cognito': {
+      // get the id_token
+      const idToken = query.id_token;
```

Comment:
`id_token` doesn't seem to exist. In the query to `auth/cognito/callback` I see the params `access_token`, `refresh_token`, `raw[access_token]`, `raw[refresh_token]`, `raw[expires_in]` and `raw[token_type]`.

Also, to do like the other providers, the code here is supposed to make a call to the cognito API to retrieve the username.

Does it work on your side with this front-end app: https://github.com/strapi/strapi-examples/tree/master/login-react ?

Reply:
you have to empty your core_store from the grant permissions as I had to update the provider oauth scope in the bootstrap file in order to retrieve the user profile information. So instead of only email we now retrieve the openid and profile scopes and you will now receive the id_token back from cognito.

Reply:
Thank, you are right and it works great with the front-end app now !

Reply:
@petersg83 and @alexandrebodin can we merge?

Reply:
Yes, it's good for me :) Just waiting for this: https://github.com/strapi/strapi/pull/6917#discussion_r499655530

Reply:
Also after that, can you update with master, squash your commits into one and signing-it off so the DCO pass?

Reply:
> Yes that is what I assumed. The access_token will be sent to frontend and then they can handle the api calls from the front end. I am now getting a 431 Request Header Fields Too Large as the access token from Cognito is quite Large more than 7K characters. How would you deal with this limitation on the server side @alexandrebodin ?

Sorry to barge in on the conversation like this, but this topic here is the only resource that's mentioning the problem occurring to me. I've tried implementing Cognito as authentication provider for Strapi, following the documentation, and using login-react as example for front-end, and I am experiencing the same error 431 Request Header Fields Too Large that you've experienced in the message above. Could you elaborate a bit more on how you solved it and what should be done to fix it? 

Steps to reproduce the error are the following: 

1. On front end, when you click on "Connect with Cognito" button you get redirected to Cognito's sign-in/sign-up form. 
2. When you choose Sign-up, create a user, and enter the verification code, apparently user gets successfully created in Cognito's User Pool ( under the hub ) but in browser, redirection happens and redirection leads to 431 Request Header Fields Too Large. 
3. When you navigate back to the URL of frontend ( ex. https://localhost:3000 ) and click again on "Connect with Cognito" you get successfully redirected to the front-end's page that's showing  successful log-in message with your Strapi's username in it.

Reply:
@rangoc this limitation seems to come from node and I think it was introduced in version 10. To overcome this I had to run the following command `node --max-http-header-size=16384 ./node_modules/strapi/bin/strapi develop` try it and let me know if that works for you.

Reply:
@ralphsomeday  Error still persists, version of Node that I am currently using is v12.16.1.

Reply:
> @ralphsomeday Error still persists, version of Node that I am currently using is v12.16.1. I think I should update my Node version to > 14 in order to be able to set --max-http-header-size=16384

Now it works 👍

Reply:
@ralphsomeday Sorry for taking this conversation in here, if it's not a good place to post this please tell me and I will delete this comment. 

I just wanted to ask, would it be possible to implement this whole flow ( the one that's implemented by react-login example ) successfully while avoiding Cognito's Hosted UI, and instead using my own custom built UI ( that I've implemented by using SDK, which enables me to authenticate with Cognito without Hosted UI )? What would be the additional steps needed to be taken for Strapi's authentication part? 

Once again, sorry for opening this topic in here, sources are scarce.

Reply:
Good question! You will have to research this one as usually the flow allows you to convert a cognito token into a strapi token which you then use to call the strapi api endpoints. If you authenticate using amplify or the sdk you will get a cognito token that won't be recognised by strapi. Maybe after getting the cognito token, you can call the redirect url to of the cognito provider with the access_token?

Reply:
Hm, sounds like an idea :) If I try and when I try I will definitely write about it. Thanks :)